### PR TITLE
BREAKING: upgrade node, pnpm, eslint shared package, dropped support for node 14

### PR DESCRIPTION
**Pull request by okkevandereijk**
### Changes
- Dropped support for `node` version 14
- Updated `node` to version 18
- Updated `pnpm` to version 8
- Shared eslint config
- Updated major semver version by using `changeset`

### CODE REVIEW

File: .changeset/early-cameras-warn.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@labdigital/commercetools-mock': major
+---
+
```
Comment (blurrah) [inline .changeset/early-cameras-warn.md:4]: I would suggest adding in a bigger notice why it's a breaking update. Something like:
`> :warning: This is a major update dropping support for Node 14 and under`

File: package.json [hunk 2]
```diff
@@ -17,10 +17,10 @@
...
   "engines": {
-    "node": ">=14",
-    "pnpm": ">=7.13.2"
+    "node": ">=16",
+    "pnpm": ">=8.1.1"
   },
-  "packageManager": "pnpm@7.13.2",
+  "packageManager": "pnpm@8.1.1",
   "scripts": {
     "start": "nodemon --watch src --exec 'node -r esbuild-register' src/server.ts",
```
Comment (blurrah) [inline package.json:25]: This can also be replaced by `tsup` in watch mode.

**Review by blurrah — CHANGES_REQUESTED** [review]

File: .changeset/early-cameras-warn.md (hunk 1 above)
Comment (okkevandereijk) [reply .changeset/early-cameras-warn.md:5]: Fixed

**Comment by okkevandereijk** [comment]
@mvantellingen The proposal is to drop support for node version 14 due to different reasons:

1. pnpm version 8+ [doesn't support node 14](https://pnpm.io/installation#compatibility) anymore
2. node 14 is EOL in[ about one month](https://github.com/nodejs/release#release-schedule)

However, quite a lot of lambda's in our projects are still running on node 14 so they are enforced to a higher lambda runtime that supports version 16+. Can you advice what to do here?

**Comment by blurrah** [comment]
I vote for dropping 14.x, older components can use 0.x and we can push for components to move to 1.x and node 18

File: package.json (hunk 2 above)
Comment (okkevandereijk) [reply package.json]: Great addition, fixed!

**Comment by mvantellingen** [comment]
Yeah let's drop nodejs 14. Unofficially it might still work i thinkg?

**Comment by blurrah** [comment]
Yeah, it will still work but you can't develop on the project with 14.x

**Comment by okkevandereijk** [comment]
> Yeah let's drop nodejs 14. Unofficially it might still work i thinkg?
In theory, yes. However I assume the `package.json` enforcements might start complaining.

**Review by blurrah — APPROVED** [review]

**Pull request merged**